src/comodo/jaxsimSimulator/jaxsimSimulator.py

```diff
@@ -0,0 +1,465 @@
+from __future__ import annotations
+
+import logging
+import math
+import pathlib
+from enum import Enum
+from typing import Any, Optional, Tuple
+
+import jax.numpy as jnp
+import jaxsim
+import jaxsim.api as js
+import numpy as np
+import numpy.typing as npt
+from jaxsim import VelRepr, integrators
+from jaxsim.mujoco import MujocoVideoRecorder
+from jaxsim.mujoco.loaders import UrdfToMjcf
+from jaxsim.mujoco.model import MujocoModelHelper
+from jaxsim.mujoco.visualizer import MujocoVisualizer
+from jaxsim.rbda.contacts.common import ContactsParams
+from jaxsim.rbda.contacts.soft import SoftContacts
+from jaxsim.rbda.contacts.relaxed_rigid import (
+    RelaxedRigidContacts,
+    RelaxedRigidContactsParams,
+)
+from jaxsim.rbda.contacts.rigid import RigidContacts, RigidContactsParams
+from jaxsim.rbda.contacts.visco_elastic import ViscoElasticContacts
+
+from comodo.abstractClasses.simulator import Simulator
+from comodo.robotModel.robotModel import RobotModel
+
+# === Logger setup ===
+logger = logging.getLogger("JaxsimSimulator")
+logger.setLevel(logging.DEBUG)
+# Remove default handlers if any
+if logger.hasHandlers():
+    logger.handlers.clear()
+logger.propagate = False
+# Console handler
+console_handler = logging.StreamHandler()
+formatter = logging.Formatter(
+    "[%(asctime)-19s.%(msecs)03d] [%(levelname)-8s] [TID %(thread)-5d] - %(message)s",
+    datefmt="%Y-%m-%d %H:%M:%S",
+)
+console_handler.setFormatter(formatter)
+logger.addHandler(console_handler)
+
+
+class ContactModelEnum(Enum):
+    RIGID = "rigid"
+    RELAXED_RIGID = "relaxed_rigid"
+    VISCO_ELASTIC = "visco_elastic"
+    SOFT = "soft"
```

Comment:
Why not using an automatic enumeration?

```suggestion
class ContactModelEnum(enum.IntEnum):
    RIGID = enum.auto()
    RELAXED_RIGID = enum.auto()
    VISCO_ELASTIC = enum.auto()
    SOFT = enum.auto()
```